packages/db/tests/models/test-granule-model.js

```diff
@@ -402,6 +402,93 @@ test('GranulePgModel.upsert() will allow a running status to replace a non-runni
   t.is(record.status, 'running');
 });
 
+test('GranulePgModel.upsert() will allow a queued status to replace a non-queued status for the same execution', async (t) => {
```

Comment:
I don't think this behavior is what we want. We only want `queued` to replace a non-queued status if it is a different execution or it is a new record. We may be more protected from this now with how you've refactored queue-granules, but our write logic still shouldn't allow it

Reply:
I'm assuming this will require edits to the conditional write logic for the PG model. And we may need to do the same thing for the Dynamo write logic